### PR TITLE
google authentication

### PR DESCRIPTION
**Pull request by GSinseswa721**
**What does this PR do?**

This PR is all about Google OAuth authentication for user sign in.

It configures Passport.js to utilize the Google OAuth strategy, allowing users to log in using their Google accounts.

**Description of Task to be completed**

Configuring Passport.js with Google OAuth strategy.
Implement routes for Google sign-in and callback handling.
Authenticate users using Google profile information.

**How should this be manually tested?**

```
npm run dev

```

### CODE REVIEW

**Comment by faid-terence** [comment]
@GSinseswa721 resolve the conflict and tests are decreasing !

**Review by elijahladdie — APPROVED** [review]

File: src/utils/auth.ts
Comment (faid-terence) [inline src/utils/auth.ts:1]: @GSinseswa721 remove the console.log(".................")

File: src/routes/ProductRoutes.ts
Comment (faid-terence) [inline src/routes/ProductRoutes.ts:1]: @GSinseswa721 Remove as any on middlewares , using any we're not utilising power of typescript.

**Review by maxCastro1 — APPROVED** [review]

File: src/routes/ProductRoutes.ts
Comment (faid-terence) [reply src/routes/ProductRoutes.ts:1]: @GSinseswa721 
Remove as any on middlewares !

**Review by faid-terence — CHANGES_REQUESTED** [review]

File: src/routes/ProductRoutes.ts
Comment (faid-terence) [inline src/routes/ProductRoutes.ts:1]: @GSinseswa721 remove unnecessary commented codes.

**Review by faid-terence — CHANGES_REQUESTED** [review]

**Review by faid-terence — APPROVED** [review]
Nice Work @GSinseswa721 !

**Pull request merged**